Lesson 3 tasks on functions and arrays

lera-kolesnik opened the pull request.

lera-kolesnik committed:
Completed first task in Lesson 3

Added an implementation in function  that takes an array of user objects and returns an array of user names using the  method

Commit:
Added task Фильтрация по диапазону

lera-kolesnik committed:
Completed second task in Lesson 3

Implementation of filterRange function using includes and logical OR operator to include elements equal to a and b in the filtered array as well

lera-kolesnik committed:
Merge branch 'master' into master

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

koalex reviewed (changes requested):
Нужно решить **3-3**

lera-kolesnik committed:
Completed third task in Lesson 3

Implement camelize function using split and join

jsru commented:
Добавляю преподавателя (@koalex) для код-ревью.

Commit:
Added task Функция showSalary(users, age)

lera-kolesnik committed:
Completed fourth task in Lesson 3

Implement showSalary function using filter, map and join methods

lera-kolesnik committed:
Merge branch 'master' into master

jsru commented:
Решение было обновлено, посмотрим что скажет @koalex

Commit:
Added task Найти min/max в строке

lera-kolesnik committed:
Redone fourth task using reduce and completed fifth task in Lesson 3

(3-4) Refactor code to use reduce instead of filter and map in showSalary function
(3-5) Add implementation in function to extract min and max numbers from a string using split, map, filter, Math.min and Math.max methods

lera-kolesnik committed:
Merge branch 'master' into master

jsru commented:
Решение было обновлено, посмотрим что скажет @koalex
